AK/BufferedStream.h

```diff
@@ -115,6 +115,73 @@ class BufferedHelper {
         return m_buffer.read(buffer);
     }
 
+    ErrorOr<StringView> read_line_with_resize(ByteBuffer& buffer)
+    {
+        return StringView { TRY(read_until_with_resize(buffer, "\n"sv)) };
+    }
+
+    ErrorOr<Bytes> read_until_with_resize(ByteBuffer& buffer, StringView candidate)
+    {
+        return read_until_any_of_with_resize(buffer, Array { candidate });
+    }
+
+    template<size_t N>
+    ErrorOr<Bytes> read_until_any_of_with_resize(ByteBuffer& buffer, Array<StringView, N> candidates)
+    {
+        if (!stream().is_open())
+            return Error::from_errno(ENOTCONN);
+
+        if (buffer.is_empty())
+            return buffer;
+
+        auto candidate = TRY(find_and_populate_until_any_of(candidates));
+
+        if (candidate.has_value()) {
+            buffer.resize(candidate->offset);
```

Comment:
I made this change and the tests broke. Even though the capacity of the buffer changes, the size does not (i.e. calling `capacity()` reflects the call but `size()` does not, so passing a buffer with an initial size of 4 will return 4 bytes. The only thing I could think of is that `resize` sets `m_size` after calling `try_ensure_capacity`, whereas calling `try_ensure_capacity` alone does not modify `m_size`. Or I could be missing something obvious.

Reply:
I mixed this up with something that actually shrinks the allocation then, sorry for that. Feel free to ignore the `resize` -> `ensure_capacity` changes.

However, proper error propagation should still be done, the non-`try` version of `resize` will simply crash if it can't allocate the necessary memory.

PS: I found that `ByteBuffer` has a `bytes_for_writing` method maybe that could end up being useful.